aQute.libg/src/aQute/lib/io/FileTree.java

```diff
@@ -0,0 +1,124 @@
+package aQute.lib.io;
+
+import static java.util.stream.Collectors.toList;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import aQute.libg.glob.AntGlob;
+
+public class FileTree {
+	private List<File>		files		= new ArrayList<>();
+	private List<Pattern>	includes	= new ArrayList<>();
+	private List<Pattern>	excludes	= new ArrayList<>();
+
+	public FileTree() {}
+
+	/**
+	 * Can be used by subclasses to add specific files to the return value of
+	 * {@link #getFiles(File, String)}.
+	 * 
+	 * @param file A file to include in the return value of
+	 *            {@link #getFiles(File, String)}.
+	 */
+	protected void addFile(File file) {
+		if (!files.contains(file)) {
+			files.add(file);
+		}
+	}
+
+	/**
+	 * Add an Ant-style glob to the include patterns.
+	 * 
+	 * @param include Add an Ant-style glob
+	 */
+	public void addInclude(String include) {
+		includes.add(AntGlob.toPattern(include));
+	}
+
+	/**
+	 * Add an Ant-style glob to the include patterns.
+	 * 
+	 * @param addIncludes Add an Ant-style glob
+	 */
+	public void addIncludes(List<String> addIncludes) {
+		for (String include : addIncludes) {
+
+			includes.add(AntGlob.toPattern(include));
+		}
+	}
+
+	/**
+	 * Add an Ant-style glob to the exclude patterns.
+	 * 
+	 * @param exclude Add an Ant-style glob
+	 */
+	public void addExclude(String exclude) {
+		excludes.add(AntGlob.toPattern(exclude));
+	}
+
+	/**
+	 * Add an Ant-style glob to the exclude patterns.
+	 * 
+	 * @param addExcludes Add an Ant-style glob
+	 */
+	public void addExcludes(List<String> addExcludes) {
+
+		for (String exclude : addExcludes) {
+
+			excludes.add(AntGlob.toPattern(exclude));
+		}
+	}
+
+	/**
+	 * Return a list of files using the specified baseDir and the configured
+	 * include and exclude Ant-style glob expressions.
+	 * 
+	 * @param baseDir The base directory for locating files.
+	 * @param defaultIncludes The default include patterns to use if no include
+	 *            patterns were configured.
+	 * @return A list of files.
+	 * @throws IOException If an exception occurs.
+	 */
+	public List<File> getFiles(File baseDir, List<String> defaultIncludes) throws IOException {
+
+
+		List<Pattern> includePatterns = includes.isEmpty() && files.isEmpty() && (!defaultIncludes.isEmpty())
```

Comment:
This check needs to be:

```java
List<Pattern> includePatterns = includes.isEmpty() && files.isEmpty() && (defaultIncludes != null)
```

This allows `null` to be passed at the defaultIncludes.  After this PR, I think I would like to add an overload of this method which is varargs:

```java
public List<File> getFiles(File baseDir, String... defaultIncludes) throws IOException {
  return getFiles(baseDir, Arrays.asList(defaultIncludes));
}
```

This would allow someone to simply call `getFiles(baseDir)` when they do not have any default includes.